src/agent0/core/hyperdrive/interactive/local_hyperdrive_test.py

```diff
@@ -1047,12 +1047,13 @@ def test_share_price_compounding_annus(fast_chain_fixture: LocalChain):
     )
     interactive_hyperdrive = LocalHyperdrive(fast_chain_fixture, interactive_config)
     hyperdrive_interface = interactive_hyperdrive.interface
+    beginning_share_price = hyperdrive_interface.current_pool_state.pool_info.lp_share_price
     logging.info(f"Variable rate: {hyperdrive_interface.current_pool_state.variable_rate}")
-    logging.info(f"Starting share price: {hyperdrive_interface.current_pool_state.pool_info.lp_share_price}")
+    logging.info(f"Starting share price: {beginning_share_price}")
     fast_chain_fixture.advance_time(YEAR_IN_SECONDS, create_checkpoints=False)
     ending_share_price = hyperdrive_interface.current_pool_state.pool_info.lp_share_price
     logging.info(f"Ending   share price: {ending_share_price}")
-    assert ending_share_price - 1 == initial_variable_rate
+    assert ending_share_price - beginning_share_price == initial_variable_rate
```

Comment:
@wakamex Can you look over this test and make sure this is still okay? This PR added a transaction between deploying the pool and returning (to register the deployed pool), so I think we can no longer assume the initial share price is 1

Reply:
lgtm if the test passes, but i'd think if starting point isn't 1 we should use

```
(ending_share_price - beginning_share_price) / beginning_share_price = initial_variable_rate
```

Reply:
A transaction actually has to be made for compounding to take place, so it's correct for the rate of return to be applied to the starting share price of 1.